### PR TITLE
Add `noindex` flag to API repsonse

### PR DESCRIPTION
**Pull request by rinsuki**
ref: #6361
image

if not local user, return value is null.

### CODE REVIEW

**Comment by Gargron** [comment]
The usefulness of this data when it's incomplete for remote users is arguable. I think for this to be done completely, the property must become part of the ActivityPub actor (and therefore move from settings to accounts table?). The property would also be a new addition to AP vocabulary and therefore warrants extended discussion.

**Comment by rinsuki** [comment]
yes, this is incomplete for remote users. however, in some cases, it is really useful. for example, auto opt-out from search engine that crawl only local timeline.

File: app/serializers/rest/account_serializer.rb [hunk 1]
```diff
@@ -48,6 +48,11 @@ def header_static
     full_asset_url(object.header_static_url)
   end
 
+  def noindex
+    user = object.user
+    user.settings['noindex'] if user
```
Comment (nightpool) [inline app/serializers/rest/account_serializer.rb]:
```ruby
return true unless object.user
object.user.settings['noindex']
```

**Comment by rinsuki** [comment]
@nightpool for example, <http://mastodonsearch.jp/>,  <https://tootsearch.chotto.moe/>, etc crawls public timeline.

**Comment by tateisu** [comment]
tootsearch reads FTL.

**Comment by hakaba-hitoyo** [comment]
The `discoverable` flag (#11688) suits for this purpose?

**Comment by Gargron** [comment]
> The `discoverable` flag (#11688) suits for this purpose?

Maybe. But `discoverable` is not the same value as "opt out of search engines". It would have slightly different meaning.

**Pull request closed without merging**